client/js/app/components/common/react_multi_select.js

```diff
@@ -0,0 +1,98 @@
+var React = require('react');
```

Comment:
I think the dependency argument here is a good one. Given that our other select components are custom, I agree the argument of following the custom pattern also has merit. I think this component is totally good to keep in here. We should probably add a few tests though.

There are probably other areas we can reduce the size of this project's build, including removing big deps like lodash in favour of modern JS native functions.